pkg/publish/options.go

```diff
@@ -16,14 +16,19 @@ package publish
 
 import (
 	"crypto/tls"
-	"log"
 	"net/http"
-	"path"
 
 	"github.com/google/go-containerregistry/pkg/authn"
-	"github.com/google/go-containerregistry/pkg/name"
 )
 
+type staticKeychain struct {
+	auth authn.Authenticator
+}
+
+func (s *staticKeychain) Resolve(resource authn.Resource) (authn.Authenticator, error) {
```

Comment:
nit: just `staticKeychain` (no `*`), since there's no situation where a call should update the value.